src/pages/blog.tsx

```diff
@@ -10,106 +10,135 @@ import { PageTitle, Text } from '../components/typography/typography';
+import { useI18next, useTranslation } from 'gatsby-plugin-react-i18next';
+import { LocalesQuery } from './index';
+import { ButtonText, SendButton } from '../components/form/controls';
+import { RightArrowIcon } from '../components/icons/buttons-icons/right-arrow';
 
 const TOP_POST_COUNT = 2;
 
 interface AllBlogPostsQuery {
-  allMarkdownRemark: {
-    nodes: {
-      id: string;
-      excerpt: string;
-      frontmatter: {
-        date: string;
-        path: string;
-        title: string;
-        featuredImage: IGatsbyImageData;
-      };
-      rawMarkdownBody: string;
-    }[];
-  };
+  nodes: {
+    id: string;
+    excerpt: string;
+    frontmatter: {
+      date: string;
+      path: string;
+      title: string;
+      featuredImage: IGatsbyImageData;
+    };
+    rawMarkdownBody: string;
+  }[];
   blogPlaceholderImage: {
     childImageSharp: {
       fluid: FluidObject;
     };
   };
 }
 
+interface BlogPageProps {
+  data: {
+    locales: LocalesQuery;
+    allMarkdownRemark: AllBlogPostsQuery;
+  };
+}
+
 const BlogPageTitle = styled(PageTitle)`
   margin-bottom: 40px;
 `;
 
 const BlogPageSubTitle = styled(Text)`
   margin-bottom: 40px;
+  white-space: pre-line;
 `;
 
-const BlogPage: React.FC = () => {
-  const data = useStaticQuery<AllBlogPostsQuery>(graphql`
-    query getBlogPosts {
-      allMarkdownRemark(
-        sort: { fields: frontmatter___date, order: DESC }
-        filter: { fileAbsolutePath: { regex: "/(blog-posts)/" } }
-      ) {
-        nodes {
-          id
-          excerpt(pruneLength: 500)
-          frontmatter {
-            date
-            path
-            title
-            featuredImage {
-              childImageSharp {
-                gatsbyImageData(
-                  width: 600
-                  aspectRatio: 1.77
-                  layout: CONSTRAINED
-                  placeholder: BLURRED
-                  formats: [AUTO, WEBP, AVIF]
-                )
-              }
-            }
-          }
-          rawMarkdownBody
-        }
-      }
-    }
-  `);
-
+const BlogPage = ({ data }: BlogPageProps) => {
+  const { t } = useTranslation();
   const blogPosts = data.allMarkdownRemark.nodes;
+  const { language, changeLanguage } = useI18next();
+
+  const onClick = (event) => {
+    event.preventDefault();
+    changeLanguage('en');
+  };
 
   return (
     <Layout light>
       <SEO title="Blog | Satellytes" />
       <Grid center>
         <GridItem>
-          <BlogPageTitle>Blog</BlogPageTitle>
+          <BlogPageTitle>{t('navigation.blog')}</BlogPageTitle>
         </GridItem>
         <GridItem md={8}>
-          <BlogPageSubTitle>
-            Our developers and designers talk about the latest trends and
-            know-how around tech, especially about our latest learnings and
-            insights on creating good software for the web.
-          </BlogPageSubTitle>
+          <BlogPageSubTitle>{t('blog.info')}</BlogPageSubTitle>
+          {language != 'en' && (
```

Comment:
The checks in the JSX - as well as the content of the checks - should be kept to a minimum, as they are quite hard to read.

In this case, you can outsource the actual content:

```
const BlogPostOverview = ({ data }) => {}
const OnlyEnglishInfo = () => {}

const BlogPage = ({ data }: BlogPageProps) => {
  return (
    <GridItem>
      <BlogPageTitle>{t('navigation.blog')}</BlogPageTitle>
    </GridItem>
    {language == 'en' ? 
       <BlogPostOverview data={data} /> : 
       <OnlyEnglishInfo />
  }
}
```